Tests/ClientRuntimeTests/NetworkingTests/DataStreamReaderTests.swift

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+@testable import ClientRuntime
+
+final class DataStreamReaderTests: XCTestCase {
```

Comment:
Can we either provide more descriptive names for the test functions? Perhaps following a format like "test_whenABC_thenXYZ"
Or add comments that describe the behavior the test is verifying

Reply:
these tests check multiple read, write seek calls so harder to name them, but I have added comments what individual step does.

Reply:
I see comments have been added to some, which are very helpful thank you! Please add to each test, since most of the tests are a bit unclear in what the expected behavior is and so I think comments would go a long way improving their maintainability moving forward. 

Two examples of improvements:

```swift
// Tests that `availableForRead` returns the correct value and
// that it decrements accordingly after each call to `read()`
func testAvailableForRead() 
```

```swift
// Tests that `contentLength` returns the correct value and 
// that the `contentLength` doesn't change after `read()` is executed.
func testContentLength()
```